src/dissect/dissect.c

```diff
@@ -370,6 +432,170 @@ static int run(int argc, char *argv[]) {
                 loop_device_relinquish(d);
                 break;
 
+        case ACTION_EXTRACT:
+        case ACTION_ADD: {
+                _cleanup_(umount_and_rmdir_and_freep) char *mounted_dir = NULL;
+                _cleanup_(rmdir_and_freep) char *created_dir = NULL;
+                _cleanup_free_ char *temp = NULL;
+
+                r = dissected_image_decrypt_interactively(
+                                m, NULL,
+                                arg_root_hash, arg_root_hash_size,
```

Comment:
If the verity metadata is passed through, ACTION_ADD would fail as the images are mounted read-only, right? So three options I can see:

1) Print a warning if any of those vars are set and ignore them
2) Print an error if ACTION_ADD and those vars are set and bail out
3) Stretch goal: use libcryptsetup to recreate the verity data after adding the file (quite a lot of work)

Reply:
Yep:

```
$ sudo buildx86/systemd-dissect img.raw
      Name: img.raw
      Size: 4.0K

Machine ID: 3d8a4aab19c64398aa48782f0a5e4ce8
OS Release: PRETTY_NAME=Debian GNU/Linux 10 (buster)
            NAME=Debian GNU/Linux
            VERSION_ID=10
            VERSION=10 (buster)
            VERSION_CODENAME=buster
            ID=debian
            HOME_URL=https://www.debian.org/
            SUPPORT_URL=https://www.debian.org/support
            BUG_REPORT_URL=https://bugs.debian.org/

RW DESIGNATOR PARTITION UUID FSTYPE   ARCHITECTURE VERITY   NODE    PARTNO
ro root       -              squashfs -            external img.raw      -
$ sudo buildx86/systemd-dissect --add img.raw /tmp/file /file
Failed to open target file '/file': Read-only file system
```

Reply:
well, you cannot add a file to a squashfs anyway. I think the error you get is actually pretty explanatory, no? it say the fs is read-only. Which it is, and it doesn't matter too much if that's because of verity or because squashfs is used or both.

I think the current behaviour makes a ton of sense: try it, and if it fails, log a good error about it, and i think the one you see is good.

Reply:
Aye, fair enough, the error is clear. I might have a use case where it would be handy to regenerate stuff - I have to repack squashfs/veritysetup it/sign it way too often, so I was going at it from that angle - having a nice tool that understands it would be cool. But I'll look into it myself, and see if it makes sense.